src/quacc/recipes/torchsim/core.py

```diff
@@ -0,0 +1,364 @@
+def relax_job(
+    atoms: list[Atoms],
+    model_type: TSModelType,
+    model_path: str | Path,
+    optimizer: Optimizer,
+    *,
+    convergence_fn: ConvergenceFn = ConvergenceFn.FORCE,
+    trajectory_reporter_dict: TrajectoryReporterDict | None = None,
+    autobatcher_dict: AutobatcherDict | bool = False,
+    max_steps: int = 10_000,
+    steps_between_swaps: int = 5,
+    init_kwargs: dict[str, Any] | None = None,
+    model_kwargs: dict[str, Any] | None = None,
+    convergence_fn_kwargs: dict[str, Any] | None = None,
+    **optimizer_kwargs: Any,
+) -> TorchSimOptSchema:
+    """
+    Carry out a geometry optimization on a set of atoms.
+
+    Parameters
+    ----------
+    atoms : list[Atoms]
+        The list of atoms objects.
+    model_type : TSModelType
+        The type of model to use, limited to the types supported by TorchSim.
+    model_path : str | Path
+        The path to the model file or checkpoint.
+    optimizer : Optimizer
+        The TorchSim optimizer to use.
+    convergence_fn : ConvergenceFn
+        The convergence function, either "energy" or "force". This will use either the
+        ts.generate_energy_convergence_fn or ts.generate_force_convergence_fn function
+        to interally generate the convergence function. Arguments can be supplied via
+        the convergence_fn_kwargs argument.
+    trajectory_reporter_dict : TrajectoryReporterDict | None
+        This dictionary defines the trajectory reporting behavior. This is a
+        quacc-specific dictionary that allows for the configuration of the
+        TrajectoryReporter. For a list of available keys, refer to the TorchSim
+        TrajectoryReporter documentation or the TrajectoryReporterDict type-hint.
+    autobatcher_dict : AutobatcherDict | bool
+        This dictionary defines the autobatcher behavior. This is a quacc-specific
+        dictionary that allows for the configuration of the autobatcher. For a list of
+        available keys, refer to the TorchSim Autobatcher's documentation or the
+        AutobatcherDict type-hint.
+    max_steps : int
+        The maximum number of steps to run for each optimization.
+    steps_between_swaps : int
+        Number of steps to take before checking convergence and swapping out states.
+    init_kwargs : dict[str, Any] | None
+        Keyword arguments passed to the optimizer initialization function.
+    model_kwargs : dict[str, Any] | None
+        Keyword arguments passed to the model.
+    convergence_fn_kwargs : dict[str, Any] | None
+        Keyword arguments passed to the convergence function generator.
+    **optimizer_kwargs : Any
+        Keyword arguments to pass to the optimizer step function.
+
+    Returns
+    -------
+    TorchSimOptSchema
+        A dictionary representing the final atoms configuration and metadata geometry
+        optimization schema. See the type-hint for the data structure.
+    """
+    runner = BaseRunner()
+    runner.setup()
```

Comment:
Here, quacc sets up a unique temporary directory where all file I/O is meant to be generated during runtime. The directory path is assigned to `runner.tmpdir`, and the base path for this unique temporary directory is controlled by the `SCRATCH_DIR` pydantic setting in quacc.